edx_api_doc_tools/view_utils.py

```diff
@@ -0,0 +1,125 @@
+"""
+Utilities for annotating API views with schema info as well as
+extracting schema info from them.
+
+External users: import these from __init__.
+"""
+from __future__ import absolute_import, unicode_literals
+
+import six
+from django.utils.decorators import method_decorator
+from drf_yasg.utils import swagger_auto_schema
+
+from .internal_utils import dedent
+
+
+def schema_for(method_name, docstring, *schema_args, **schema_kwargs):
+    """
+    Decorate a class to specify a schema for one of its methods.
+
+    Useful when the method you are describing is not defined inside of your
+    class body, but is instead defined somewhere up in the DRF view hierarchy.
+    (For applying a schema directly to a method, use `schema`).
+
+    Valid values of `method_name` include:
+        'list',
+        'get', 'GET', 'retrieve',
+        'post', 'POST', 'create',
+        'put', 'PUT',
+        'patch', 'PATCH', 'update',
+        'delete', 'DELETE', and 'destroy'.
```

Comment:
Why does this decorator care what the method name is?  Maybe this should say, "DRF method names include:"

Reply:
It doesn't, but `swagger_auto_schema` only works on those names. Regardless, I like your rewording.